### ISSUE TITLE
croc-hunter does not work when routed thru the proxy

### ISSUE THREAD

**Issue by goffinf** [vfarcic/docker-flow-proxy#49]
Docker version 1.12.1, build 23cf638, experimental
Image: lachlanevenson/croc-hunter:v1

```
$ docker node ls
ID                           HOSTNAME  STATUS  AVAILABILITY  MANAGER STATUS
09hn33u8wfa3xnq87qm6eisz3    swarm-2   Ready   Active        Leader
41cmpzi5lia59czv4xquv5hxg    swarm-3   Ready   Active        Reachable
972vehz62isj9khbcsrh1mgb2 *  swarm-1   Ready   Active        Reachable

$ docker-machine ls
NAME         ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER    ERRORS
puppetTest   -        virtualbox   Stopped                                       Unknown
swarm-1      *        virtualbox   Running   tcp://192.168.99.100:2376           v1.12.1
swarm-2      -        virtualbox   Running   tcp://192.168.99.101:2376           v1.12.1
swarm-3      -        virtualbox   Running   tcp://192.168.99.102:2376           v1.12.1
```

If I run croc-hunter as a stand-alone container on a given node (say: 192.168.99.100) everything works fine ...

`docker run -d --name crock-hunter -p 9080:8080 lachlanevenson/croc-hunter:v1`

Browse to http://192.168.99.102:9080/

game is visable and plays correclty ...

![croc-hunter-ok](https://cloud.githubusercontent.com/assets/2772752/19219828/8d2945e6-8e15-11e6-95c1-ae59d325a800.jpg)

If I configure croc-hunter as a swarm service and route thru docker-flow-proxy it would appear that the IP:port resolves correctly but the game is NOT visible ?

```
docker service create --replicas 1 --name croc-hunter --network proxy lachlanevenson/croc-hunter:v1

curl "$(docker-machine ip swarm-1):8080/v1/docker-flow-proxy/reconfigure?serviceName=croc-hunter&servicePath=/croc&port=8080"

$ docker service ls
ID            NAME         REPLICAS  IMAGE                          COMMAND
1ki8tcds88zj  croc-hunter  1/1       lachlanevenson/croc-hunter:v1
78qhls4bo00a  go-demo-db   1/1       mongo
8yekl0g804u1  proxy        1/1       vfarcic/docker-flow-proxy
f0qfprgyc7se  go-demo      1/1       vfarcic/go-demo
```

Then browse to http://192.168.99.100/croc

The browser tab show Croc Hunter and the page source is exactly the same as the stand-alone container, BUT the app is NOT visible ...

![croc-hunter-proxy](https://cloud.githubusercontent.com/assets/2772752/19219888/83c1caf8-8e17-11e6-87c9-70d2b4b490e5.jpg)

![croc-hunter-source](https://cloud.githubusercontent.com/assets/2772752/19219889/88639866-8e17-11e6-8e9e-e79385478823.jpg)

**Comment by vfarcic** [comment]
The problem is in the `servicePath` of the reconfigure request. HAProxy uses that value to distinguish requests. Imagine, that you have two services configured as follows.

```bash
curl "$(docker-machine ip swarm-1):8080/v1/docker-flow-proxy/reconfigure?serviceName=my-service-1&servicePath=/service1&port=8080"
```

**Comment by goffinf** [comment]
Thx Viktor, I had tried that and in my case it did NOT work, or rather I got a NOK response so I had (wrongly) assumed that I wouldn't be able to use a servicePath that just has /

Once I realised that was because I hadn't created the service yet and then did so, I tried again and of course the config entry WAS succesfully created and the croc-hunter game runs properly.

This perhaps speaks to the discussion points I made about this container in relation to your [excellent] DevOps Toolkit 2.1 here - https://disqus.com/home/discussion/channel-thedevops21toolkit/chapter_service_discovery_inside_a_swarm_cluster/

In particular, it appears that ...

a. If you have not CREATED the service BEFORE attempting to add the service configuration to docker-flow-proxy then you will receive a NOK. Personally I think this would be a useful feature so I'll raise that separately although I can understand if you would prefer not to implement it since there is perhaps an equally persuasive case for insisting that the service exists. But ultimately this was probably the root cause of [my] confusion that created this particular issue.

b. If you explicitly REMOVE a service the config entry does NOT appear to be removed from docker-flow-proxy. That seems a bit inconsistent with (a) to me ?

c. If you want to stick with the current behaviour (i.e. service must exist) then perhaps the NOK response could make that condition more obvious (and the docs). I'll raise this as a separate issue for you to consider.

c. To make it easier to check what is / is not configured rather than having to hunt down where the docker-flow-proxy container is running and look at its haproxy.cfg file, it would be really useful for a service (list ?  possibly with the ability to regEx filter on serviceName) to be added to the API. I'll raise that as a separate issue too.

Thx again Viktor.
